Assets/Scripts/EndGoal.cs

```diff
@@ -8,28 +8,34 @@ public class EndGoal : MonoBehaviour
 {
 
     private string frogsRemainingText;
-    private int frogGoal = 1;
-
-
-    void Update()
+    private int frogGoal = 0;
+    private int frogAmount = 0; //amount of frogs in level
+    public int FrogGoal 
     {
-        frogsRemainingText = $"Frogs: {frogGoal}/{FrogSpawner.amountOfFrogs}";
+            get { return frogGoal; }
     }
 
+    private ParticleSystem goalParticles; //confetti particle system
+
     //observer pattern that tells frog remaining UI how many frogs left
-    public static event Action<string> FrogReachesGoalTriggered; 
+    public static event Action<string> FrogReachesGoalTriggered;
+
+    private void Start()
+    {
+        frogAmount = FrogSpawner.amountOfFrogs;
+        goalParticles = GetComponentInChildren<ParticleSystem>();
+        frogsRemainingText = $"Frogs: {frogGoal}/{frogAmount}";
+    }
 
     private void OnTriggerEnter2D(Collider2D other)
     {
         if(other.gameObject.tag == "Frog")
         {
+            goalParticles.Play(); //launches confetti
             Destroy(other.gameObject); //destroys frog
             frogGoal++;
 
-            if (frogGoal == FrogSpawner.amountOfFrogs + 1)
-            {
-                frogsRemainingText = "You Win!";
-            }
+            frogsRemainingText = $"Frogs: {frogGoal}/{frogAmount}";
 
             if (FrogReachesGoalTriggered != null)
```

Comment:
you can use the ?. syntax here to simplify and remove the if statement. https://thedotnetguide.com/null-conditional-operator-in-csharp/#

`FrogReachesGoalTriggered?.Invoke()`